arch/arc/core/CMakeLists.txt

```diff
@@ -1,4 +1,6 @@
-zephyr_sources(
```

Comment:
The commit message should explain why we are re-organizing from the zephyr_* API to the zephyr_library_* API.

(I know why, so it is not necessary to explain this to me)